FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/tls_freertos.c

```diff
@@ -0,0 +1,611 @@
+                        mbedtlsHighLevelCodeOrDefault( mbedtlsError ),
+                        mbedtlsLowLevelCodeOrDefault( mbedtlsError ) ) );
+
+            returnStatus = TLS_TRANSPORT_HANDSHAKE_FAILED;
+        }
+    }
+
+    if( returnStatus != TLS_TRANSPORT_SUCCESS )
+    {
+        sslContextFree( &( pNetworkContext->sslContext ) );
+    }
+    else
+    {
+        LogInfo( ( "(Network connection %p) TLS handshake successful.",
+                   pNetworkContext ) );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static TlsTransportStatus_t initMbedtls( void )
+{
+    TlsTransportStatus_t returnStatus = TLS_TRANSPORT_SUCCESS;
+    int mbedtlsError = 0;
+
+    /* Set the mutex functions for mbed TLS thread safety. */
+    mbedtls_threading_set_alt( mbedtls_platform_mutex_init,
+                               mbedtls_platform_mutex_free,
+                               mbedtls_platform_mutex_lock,
+                               mbedtls_platform_mutex_unlock );
+
+    /* Initialize contexts for random number generation. */
+    mbedtls_entropy_init( &entropyContext );
+    mbedtls_ctr_drbg_init( &ctrDrgbContext );
+
+    /* Add a strong entropy source. At least one is required. */
+    mbedtlsError = mbedtls_entropy_add_source( &entropyContext,
+                                               mbedtls_platform_entropy_poll,
+                                               NULL,
+                                               32,
+                                               MBEDTLS_ENTROPY_SOURCE_STRONG );
+
+    if( mbedtlsError != 0 )
+    {
+        LogError( ( "Failed to add entropy source: mbedTLSError= %s : %s.",
+                    mbedtlsHighLevelCodeOrDefault( mbedtlsError ),
+                    mbedtlsLowLevelCodeOrDefault( mbedtlsError ) ) );
+        returnStatus = TLS_TRANSPORT_INTERNAL_ERROR;
+    }
+
+    if( returnStatus == TLS_TRANSPORT_SUCCESS )
+    {
+        /* Seed the random number generator. */
+        mbedtlsError = mbedtls_ctr_drbg_seed( &ctrDrgbContext,
+                                              mbedtls_entropy_func,
+                                              &entropyContext,
+                                              NULL,
+                                              0 );
+
+        if( mbedtlsError != 0 )
+        {
+            LogError( ( "Failed to seed PRNG: mbedTLSError= %s : %s.",
+                        mbedtlsHighLevelCodeOrDefault( mbedtlsError ),
+                        mbedtlsLowLevelCodeOrDefault( mbedtlsError ) ) );
+            returnStatus = TLS_TRANSPORT_INTERNAL_ERROR;
+        }
+    }
+
+    if( returnStatus == TLS_TRANSPORT_SUCCESS )
+    {
+        LogDebug( ( "Successfully initialized mbedTLS." ) );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+TlsTransportStatus_t TLS_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
```

Comment:
These functions seem to be public but undocumented in this file, I expected to see a Doxygen style header for these explaining the usage?

Reply:
Yes. These are already has doxygen style header in their header file(tls_freertos.h)